src/foam/nanos/auth/LanguageChoiceView.js

```diff
@@ -0,0 +1,237 @@
+    'foam.nanos.auth.Language',
+  ],
+
+  imports: [
+    'stack',
+    'languageDAO',
+    'subject',
+    'userDAO',
+    'countryDAO',
+    'translationService'
+  ],
+
+  exports: [ 'as data' ],
+
+  css: `
+  ^carrot {
+    width: 0;
+    height: 0;
+    border-left: 5px solid transparent;
+    border-right: 5px solid transparent;
+    border-top: 5px solid white;
+    display: inline-block;
+    float: right;
+    margin-top: 7px;
+    margin-left: 7px;
+  }
+  ^ .foam-u2-ActionView-languageChoice {
+    display: inline-block;
+    background: none !important;
+    border: 0 !important;
+    box-shadow: none !important;
+    width: max-content;
+    cursor: pointer;
+    margin-right: 27px;
+  }
+  ^ .foam-nanos-u2-navigation-TopNavigation-LanguageChoiceView {
+    align-items: center;
+  }
+  ^ .foam-u2-ActionView-languageChoice > span {
+    font-family: /*%FONT1%*/ Roboto, 'Helvetica Neue', Helvetica, Arial, sans-serif;
+    font-size: 16px;
+    font-weight: 300;
+    letter-spacing: 0.2px;
+    color: #ffffff;
+  }
+  ^ .popUpDropDown > div {
+    width: 100%;
+    text-align: center;
+    height: 25;
+    padding-bottom: 5;
+    font-size: 14px;
+    font-weight: 300;
+    letter-spacing: 0.2px;
+    color: /*%BLACK%*/ #1e1f21;
+    line-height: 30px;
+  }
+  ^ .foam-u2-PopupView {
+    left: -30 !important;
+    top: 51px !important;
+    padding: 0 !important;
+    z-index: 1000;
+    width: 110px !important;
+    background: white;
+    opacity: 1;
+    box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.19);
+  }
+  ^ .popUpDropDown > div:hover{
+    background-color: #1cc2b7;
+    color: white;
+    cursor: pointer;
+  }
+  ^ .popUpDropDown::before {
+    content: ' ';
+    position: absolute;
+    height: 0;
+    width: 0;
+    border: 8px solid transparent;
+    border-bottom-color: white;
+    -ms-transform: translate(110px, -16px);
+    transform: translate(50px, -16px);
+  }
+  ^ .flag {
+    width: 30px !important;
+    height: 17.6px;
+    object-fit: contain;
+    padding-top: 6px;
+    padding-left: 10px;
+    margin-right: 23px;
+  }
+  ^ img {
+    height: 17.6px !important;
+    margin-right: 6;
+    width: auto;
+  }
+  `,
+
+  properties: [
+    'optionsBtn_',
+    'supportedLanguages',
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.auth.Language',
+      name: 'defaultLanguage',
+      factory: function() {
+        return foam.nanos.auth.Language.create({code: 'en'})
+      }
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.auth.Language',
+      name: 'lastLanguage',
+      factory: function() {
+        let language = this.supportedLanguages.find( e => e.toString() === foam.locale )
+        language = language === undefined ? this.defaultLanguage : language
+        localStorage.setItem('localeLanguage', language.toString());
+        return language
+      }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this.updateE()
+    },
+    async function updateE() {
```

Comment:
Why is this not just in initE()?

Reply:
initE() can not use async

Reply:
Why not?

Reply:
I got error when I have async in initE previously. But I forgot what the reason is.  I will try again, see if it will work with async in initE.